core/infrastructure/marketplace-api-client-adapter/adapters/billing-profile-client-adapter.ts

```diff
@@ -54,4 +65,81 @@ export class BillingProfileClientAdapter implements BillingProfileStoragePort {
       tag,
     };
   };
+
+  getBillingProfileInvoicePreviewById = ({
+    pathParams,
+    queryParams,
+    impersonationHeaders,
+  }: FirstParameter<BillingProfileStoragePort["getBillingProfileInvoicePreviewById"]>) => {
+    const path = this.routes["getBillingProfileInvoicePreviewById"];
+    const method = "GET";
+    const tag = HttpClient.buildTag({ path, pathParams, queryParams });
+    const request = async () => {
+      const data = await this.client.request<GetBillingProfileInvoicePreviewByIdResponse>({
+        path,
+        method,
+        tag,
+        pathParams,
+        queryParams,
+        headers: {
+          accept: "application/pdf",
+          ...(impersonationHeaders ? { "X-Impersonation-Claims": impersonationHeaders } : {}),
```

Comment:
Impersonation is already handled inside `http-client`

Reply:
right!